index.js

```diff
@@ -22,24 +24,16 @@ function GitHub(options) {
     return new GitHub(options);
   }
 
-  this.options = options || {};
-  this.defaults = utils.defaults(this.options);
-
-  if (!this.options.apiurl) {
-    this.options.apiurl = 'https://api.github.com';
-  }
-
-  this.interpolate = function(opts) {
-    var data = extend({}, this.options, opts);
-    return utils.interpolate(opts.path, data);
-  }.bind(this);
+  this.options = typeof options === 'object' ? options : {};
```

Comment:
change this to

```js
options = options || {}
```
of do `if (!options) options = {}`, but `typeof options === 'object'` won't catch `null`